executor/tests/test_executor.py

```diff
@@ -75,6 +94,31 @@ def test_send_message_max_length_exceeded(self):
         result = ce.send_message(driver, message, max_message_length)
         self.assertFalse(result)
 
+    def test_progress_group_assignment_and_killing(self):
+
+        start_time = time.time()
+
+        task_id = get_random_task_id()
+        command = 'echo "A.$(sleep 30)" & echo "B.$(sleep 30)" & echo "C.$(sleep 30)" &'
+        task = {'task_id': {'value': task_id},
+                'data': encode_data(json.dumps({'command': command}).encode('utf8'))}
+        environment = {}
+        process = ce.launch_task(task, environment)
+
+        group_id = ce.find_process_group(process.pid)
+        self.assertGreater(group_id, 0)
+
+        child_process_ids = find_process_ids_in_group(group_id)
+        self.assertGreaterEqual(len(child_process_ids), 7)
+
+        ce.kill_process_group(group_id)
+        time.sleep(1)
+
+        child_process_ids = find_process_ids_in_group(group_id)
+        self.assertLessEqual(len(child_process_ids), 1)
+
+        self.assertLess(time.time() - start_time, 30)
```

Comment:
What is this for?

Reply:
I assert less than 30 seconds to ensure the kill took place without having to wait for the 30 seconds for `sleep` to complete.